Add option for configure a custom "turn_off_action" (lg_netcast integration)

ChrisHaPunkt opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This update for the lg_netcast integration adds the possiblity to override the default "tv off" option with a custom service call (exactly like the already implemented "tv on" action service call). 
This is useful, if you want to handle the power on and off calls with a smart power plug and not with the native "tv off" functionality which turns the tv in standby.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml
  - platform: lg_netcast
    access_token: <token>
    host: 192.168.1.23
    turn_on_action:
      service: switch.turn_on
      data:
        entity_id: switch.my_tv
    turn_off_action:
      service: switch.turn_off
      data:
        entity_id: switch.my_tv
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/13018

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

ChrisHaPunkt committed:
Add option for configure a custom "turn_off_action"

homeassistant commented:
Hi @ChrisHaPunkt, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#34301).

Once you do that we will be able to review and accept this pull request.

Thanks!

ChrisHaPunkt committed:
reformatted by black

raman325 committed:
Refactor vizio media_player tests to remove conditional statements from helper function (#33615)

raman325 committed:
Skip ignored hosts when checking existing config entries in config flow (#34280)

basnijholt committed:
Pass an argument to kef.update_dsp for async_track_time_interval (#34310)

bdraco committed:
Use config entry id for unique id if serial number is missing (#34154)

Kane610 committed:
UniFi - Support automatic removal of clients (#34307)

amelchio committed:
Purge recorder database at night (#33646)

amelchio committed:
Fixes for Sonos media titles (#34311)

bdraco committed:
Fix default elkm1 temp units (#34274)

bdraco committed:
Resolve homekit not updating motion sensors (#34282)

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

raman325 committed:
Abort vizio zeroconf config flow if unique ID is already configured (#34313)

balloob committed:
Update Coordinator: Only schedule a refresh if listenerrs (#34317)

danielperna84 committed:
Update pyhomematic to 0.1.66 (#34314)

homeassistant committed:
[ci skip] Translation update

jnimmo committed:
Bump to pyIntesisHome 1.7.4 (#34319)

bdraco committed:
Restore isy light brightness after off (#34320)

balloob committed:
Improve error message when people have not moved config flow title yet (#34321)

Kane610 committed:
UniFi - Allow tracking of clients connected to third party APs (#34067)

* Allow disable wired bug work around

* Move small improvements from closed PR #34065

* Fix failing test

* Add new test

* Some extra logging

* Harmonize log outputs

* Add config flow string

* Fix Balloobs comments

balloob committed:
Remove two more titles from strings.json (#34324)

janiversen committed:
Rollback modbus to version 0.107.7 keep new functionality (#34287)

* Rollback modbus to version 0.107.7

Update manifest to not use async.

Rollback entities to sync version.

Keep newer modifications apart from async.

Rollback __init__ to sync version but keep the new functionality.

add async sub directory

Adding the current (not working) version in a sub directory,
to allow easy sharing with a few alfa testers.

The async version are to be updated to use the serial/tcp already
available instead of the flaky pymodbus version. pymodbus is still
needed to encode/decode the messagess.

Update test cases to reflect sync implementation, but
keep the new functionality like e.g. conftest.py.

* do not publish async version

The async version will be made available in a forked repo, until
it is ready to replace the production code.

ziv1234 committed:
Fix uncaught exceptions in ios (#34119)

* verify that the config in hass is not empty

* changed to use MockConfigEntry

* Update tests/components/ios/test_init.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/ios/test_init.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* changed the test per suggestions

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
